cmake/modules/extensions.cmake

```diff
@@ -2094,6 +2098,30 @@ function(target_cc_option_fallback target scope option1 option2)
   endif()
 endfunction()
 
+function(target_ld_libraries target scope)
```

Comment:
the existing `target_ld_options()` tries to verify that an option passed to the linker is indeed valid by the compiler (which invokes the linker).

That make sense for linker flags, but it doesn't make sense for a `-l<library>`, cause the `-l` itself is always valid, but whether the lib can be linked in depends on the library paths. 

So why this extra check if creating a dedicated `target_ld_libraries()` function ?

In addition, if we allow linking of libraries this way, then I guess it makes sense to omit the `-l` from the caller and let this function append `-l`.

That will even prevent callers from misusing `target_ld_libraries()` to pass linker options.

Reply:
Oh, I assumed the check would also verify that the library existed, like `-ldl` as used in the posix backend?
Otherwise, yes, having it add `-l` to every option passed would avoid having it misused.

Reply:
mostly, I didn't want to change *how* the options were validated, just the function used to get them into the link command

Reply:
> Oh, I assumed the check would also verify that the library existed

how can it do that if it doesn't have all library path specified ?
Sure for libs available in default path, this would be possible, for other libs not in path, then this is not working.

At this stages we want to check if a linker flag (`-l`) is valid, not if the filibrary can be found.
The linking stage itself we provide info on missing libraries for linking.

Reply:
that makes sense. And, we assume that the `-l` flag is valid because it's a linker. Sounds like it's a much simpler process in this case.